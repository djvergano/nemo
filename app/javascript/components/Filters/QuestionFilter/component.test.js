import React from 'react';
import { shallow } from 'enzyme';

import { getFiltersStore } from '../testUtils';

import Component from './component';
import { quietMount } from '../../../testUtils';

jest.mock('../../conditions/ConditionSetFormField/component', () => 'ConditionSetFormField');
jest.mock('../../conditions/AddConditionLink/component', () => 'AddConditionLink');

const defaultProps = {
  filtersStore: getFiltersStore(),
  onSubmit: jest.fn(),
};

it('renders as expected', () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

describe('popover', () => {
  const wrapper = quietMount(<Component {...defaultProps} />);
  wrapper.find('Button#question-filter').simulate('click');

  const overlay = quietMount(wrapper.find('OverlayTrigger').prop('overlay'));

  it('handles callbacks', () => {
    overlay.find('Button.btn-apply').simulate('click');
    expect(defaultProps.onSubmit.mock.calls.length).toMatchSnapshot();
    expect(defaultProps.onSubmit.mock.results).toMatchSnapshot();
  });
});
