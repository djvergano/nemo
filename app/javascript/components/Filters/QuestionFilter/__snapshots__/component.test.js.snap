// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover handles callbacks 1`] = `1`;

exports[`popover handles callbacks 2`] = `
Array [
  Object {
    "type": "return",
    "value": undefined,
  },
]
`;

exports[`renders as expected 1`] = `
<FilterOverlayTrigger
  buttonClass="btn-margin-left"
  buttonsContainerClass="condition-margin"
  hints={
    Array [
      "One",
    ]
  }
  id="question-filter"
  onSubmit={[MockFunction]}
  popoverClass="wide display-logic-container"
  popoverContent={
    <div>
      <p
        className="mb-2"
      >
        Showing questions from Two only.
      </p>
      <ConditionSetFormField />
      <AddConditionLink />
    </div>
  }
  title="Question"
/>
`;
